File: careerforge/src/main/java/com/careerforge/suggestion/entity/Suggestion.java
package com.careerforge.suggestion.entity;

import com.careerforge.common.entity.BaseEntity;
import com.careerforge.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "suggestions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Suggestion extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id", updatable = false, nullable = false)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id")
    private User student;

    @Column(name = "content", nullable = false, columnDefinition = "TEXT")
    private String content;

    @Column(name = "reply_content", columnDefinition = "TEXT")
    private String replyContent;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "replied_by")
    private User repliedBy;

    @Column(name = "replied_at")
    private LocalDateTime repliedAt;
}

File: careerforge-frontend/src/types/suggestion.ts
export interface Suggestion {
  id: string;
  studentId: string | null;
  studentName: string;
  content: string;
  replyContent: string | null;
  repliedById: string | null;
  repliedByName: string | null;
  createdAt: string;
  repliedAt: string | null;
}

export interface CreateSuggestionRequest {
  content: string;
}

export interface ReplySuggestionRequest {
  replyContent: string;
}
